crash/session.py

```diff
@@ -29,7 +30,42 @@ def __init__(self, kernel_exec=None, vmcore=None, kernelpath=None,
         if not kernel_exec:
             return
 
-        error = gdb.execute("file {}".format(kernel_exec), to_string=True)
+        try:
+            kdump = kdumpfile(vmcore)
+        except OSErrorException as e:
+            raise RuntimeError(str(e))
+
+        kaslr_off = long(
+            kdump.attr.get("linux.vmcoreinfo.lines.KERNELOFFSET", "0"),
+            base=16)
+
+        error = gdb.execute("exec-file {}".format(kernel_exec), to_string=True)
+        tinfo = gdb.execute("info target", to_string=True)
+        args=""
+        textaddr = 0
+        in_exec = False
+        for line in tinfo.splitlines():
+            if line.startswith((" ", "\t")):
+                if not in_exec:
+                    continue
+                try:
+                    (addrs, sect) = line.strip().split(" is ")
+                except ValueError:
+                    continue
+                (start, end) = addrs.split(" - ")
+                sect = sect.split(' ')[0]
+                startaddr = long(start, base=0) + kaslr_off
+                if sect == ".text":
+                    textaddr = startaddr
+                    args += " 0x{:x}".format(startaddr)
+                elif startaddr >= textaddr:
```

Comment:
What does this condition translate to? In my (non KASLR) ELF it'd only apply to `.data..percpu`.

Reply:
Indeed, this serves only the percpu section for now. I'm trying to prepare for a potential rename. The .text section is always the first section in a kernel image, so this looks robust enough to me. Note that the "_stext" symbol is sometimes used to find the beginning of the kernel; it's even exported through VMCOREINFO for that purpose.